File: psychology_backend/src/main/java/com/psychology/services/AdminService.java
package com.psychology.services;

import java.io.IOException;
import java.time.LocalDate;
import java.util.List;

import org.springframework.web.multipart.MultipartFile;

import com.psychology.dtos.AnaliticDTO;
import com.psychology.dtos.PsychologistDTO;
import com.psychology.dtos.SpecializationDTO;
import com.psychology.entities.Call;
import com.psychology.entities.Office;
import com.psychology.entities.Psychologist;
import com.psychology.entities.Record;
import com.psychology.entities.Specialization;
import com.psychology.exceptions.NotFoundException;
import com.psychology.exceptions.PsychologistAlreadyExists;
import com.psychology.exceptions.SpecializationAlreadyExists;

import jakarta.transaction.Transactional;

public interface AdminService {
	Iterable<Psychologist> getPsychologists();

	Iterable<Specialization> getSpecializations();

	Iterable<Office> getOffices();
	
	Iterable<Record> getRecords();
	
	Iterable<Call> getCalls();

	Integer deleteSpecialization(Integer id) throws NotFoundException;

	Specialization addSpecialization(SpecializationDTO specializationDTO) throws SpecializationAlreadyExists;

	@Transactional
	Psychologist addPsychologist(PsychologistDTO psychologistDTO, MultipartFile[] files) throws PsychologistAlreadyExists, IOException, NotFoundException;
	
	@Transactional
	Psychologist editPsychologist(PsychologistDTO psychologistDTO, MultipartFile[] files) throws PsychologistAlreadyExists, IOException, NotFoundException;
	
	Integer deletePsychologist(Integer id);
	
	Integer deleteRecord(Integer id);
	
	Integer callBack(Integer id) throws NotFoundException;
	
	Psychologist getPsychologistById(Integer id) throws NotFoundException;
	
	List<AnaliticDTO> getAnalitic(LocalDate date);
}

File: psychology_frontend/src/API/AdminService.js
import axios from "axios";
import getAuthHeader from "utils/getAuthHeader";
const API_URL = "http://localhost:8080/api/admin/";

export default class AdminService {
  static async getPsychologists() {
    const response = await axios.get(API_URL + "psychologists", {
      headers: getAuthHeader(),
    });
    return response;
  }

  static async getSpecializations() {
    const response = await axios.get(API_URL + "specializations", {
      headers: getAuthHeader(),
    });
    return response;
  }

  static async getOffices() {
    const response = await axios.get(API_URL + "offices", {
      headers: getAuthHeader(),
    });
    return response;
  }

  static async getRecords() {
    const response = await axios.get(API_URL + "records", {
      headers: getAuthHeader(),
    });
    return response;
  }

  static async getAnalitic(date) {
    const response = await axios.get(API_URL + "analitic", {
      headers: getAuthHeader(),
      params: {
        date: date,
      },
    });
    return response;
  }

  static async getCalls() {
    const response = await axios.get(API_URL + "calls", {
      headers: getAuthHeader(),
    });
    return response;
  }

  static async deleteSpecializationById(id) {
    const response = await axios.delete(API_URL + "specialization/" + id, {
      headers: getAuthHeader(),
    });
    return response;
  }

  static async deleteRecordById(id) {
    const response = await axios.delete(API_URL + "record/" + id, {
      headers: getAuthHeader(),
    });
    return response;
  }

  static async callBackById(id) {
    const response = await axios.get(API_URL + "callBack/" + id, {
      headers: getAuthHeader(),
    });
    return response;
  }

  static async addSpecialization({ specialization }) {
    const response = await axios.post(
      API_URL + "addSpecialization",
      specialization,
      {
        headers: getAuthHeader(),
      }
    );
    return response;
  }

  static async addPsychologist(formData) {
    const response = await axios.post(API_URL + "addPsychologist", formData, {
      headers: getAuthHeader(),
    });
    return response;
  }

  static async editPsychologist(formData) {
    const response = await axios.patch(API_URL + "editPsychologist", formData, {
      headers: getAuthHeader(),
    });
    return response;
  }

  static async deletePsychologistById(id) {
    const response = await axios.delete(API_URL + "psychologist/" + id, {
      headers: getAuthHeader(),
    });
    return response;
  }

  static async getPsychologistById(id) {
    const response = await axios.get(API_URL + "psychologist/" + id, {
      headers: getAuthHeader(),
    });
    return response;
  }
}
